Indice changes

basebala opened the pull request:
Added a new eval-mode such that developers can input indices of images from dataset (zero-indexed) manually as the evaluation dataset.

To use, one can use command 'nerfstudio-data --eval-mode indices --eval-image-indices [spaced list of image indices for evaluation]' with their 'ns-train' command.

Any image with an index not specified in --eval-image-indices will be used for training.

Commit:
added functionality for indices eval-mode

Commit:
Merge branch 'main' of https://github.com/nerfstudio-project/nerfstudio into indice_changes

Commit:
reformat

akristoffersen reviewed (approved):
This is awesome! Seems super useful, thanks for contributing.

basebala committed:
Merge branch 'main' into indice_changes

basebala committed:
Change error from array to float in test cases

basebala committed:
Update Dockerfile

basebala committed:
Modify tiny-cuda-nn installation command in Dockerfile

Updated pip install command for tiny-cuda-nn to include --no-build-isolation.

basebala committed:
Update Dockerfile to install specific torch versions

basebala committed:
Enhance Dockerfile with package installations

Updated the Dockerfile to install additional Python packages and upgrade setuptools.

basebala committed:
Update pip install command for gsplat package

basebala commented:
@akristoffersen can we merge?
